### ISSUE TITLE
Option to use JIRA issue name instead of issue key for output file names

### ISSUE THREAD

**Issue by serdarbuyuran** [Xray-App/xray-maven-plugin#68]
Hello,

I am currently using the `mvn xray:export-features` command to export features and scenarios from JIRA using the Xray plugin. The output file names are in the format `1_XRAY-3115.feature` where `XRAY-3115` is the issue key.

However, I want to replace the issue key in the file name with the corresponding JIRA issue name. In other words, instead of `1_XRAY-3115.feature`, I want the file name to represent the actual JIRA issue name. This would provide a more meaningful context when looking at the output files.

Could you consider adding a feature that allows the customization of the output file names in this way?

Thank you for considering my request.

**Comment by bitcoder** [comment]
Hi @serdarbuyuran ,
currently this plugin gets the generated features from Xray API; so it's Xray that generates the name according with the logic in the product, which has some rules if you look at Xray's documentation.
Implementing your feature would be hard for several reasons; the main one is that it would require making API call to Jira's REST API to obtain it after parsing the names that  come in the zipped content given by Xray...

**Comment by orcunbalcilar** [comment]
@bitcoder you are right, but there is already a feature name at the top of each feature file - > Feature: Regression. So we can get this name and rename each feature file I think as a lightweight and alternative solution.
This enhancement would be beneficial because xray issue keys makes no sense in terms of naming feature files.

**Comment by sandeepraina-lenze** [comment]
Hello,

I just checked the same feature with current build and it works provided a simple setting/configuration required to be done in JIRA under Managed Apps for XRay

![image](https://github.com/user-attachments/assets/31398d37-d7c4-4127-af4d-25ebcd20db13)
